Removing Unecessary Animation on Clear Button

marchiore opened the pull request:
@oliviabrown9  as you requested, continuation of #1257

oliviabrown9 commented:
@marchiore Thanks for opening a new PR! I'm not confident that adding in a time delay is the best solution. It feels a bit like a hack rather than getting to the root of the problem Try to look at why it animates even when you hide it until that animation has completed? Is there another animation happening? What constraints are causing this?

marchiore commented:
@oliviabrown9 You have a point there!

I checked the constraints and i noticed that when the user start editing the urlBar text, the app activate the **pageActionsButton** constraints that's where the clear button animation is coming from.

So, to debug it, i did:

- Changed `urlText.rightViewMode` from **.whileEditing** to  **.always**, now i can see what's going on and i noticed that the **urlText** trailing contrains are connected to **pageActionsButton** leading by the code:

`make.trailing.equalTo(pageActionsButton.snp.leading)`

![screen shot 2018-08-25 at 12 14 06](https://user-images.githubusercontent.com/648079/44619745-ae0a3a00-a861-11e8-8b7e-4e127d8c4942.png)

So, i changed it to:

`make.trailing.equalTo(textAndLockContainer)`

And i got this result:

![screen shot 2018-08-25 at 12 28 23](https://user-images.githubusercontent.com/648079/44619899-6df78700-a862-11e8-90ee-e2f90a518272.png)

So now i changed again `urlText.rightViewMode` from **.always** to **.whileEditing** and now it has  no animation while editing.

![ezgif-2-a80aaf6284](https://user-images.githubusercontent.com/648079/44620005-19eda200-a864-11e8-9609-b9fe69f4d98a.gif)

But now i noticed another problem and i'm a few hours trying to solve this and if you can give me some advice i'll really appreciate :)

When i open a big url like:

[http://www.thelongestdomainnameintheworldandthensomeandthensomemoreandmore.com/](http://www.thelongestdomainnameintheworldandthensomeandthensomemoreandmore.com/)

the **urlBar text** overlap **pageActionsButton** like the image below.

![screen shot 2018-08-25 at 12 17 01](https://user-images.githubusercontent.com/648079/44619946-5bca1880-a863-11e8-945c-7b96c1dd9bc4.png)

Any advice how can i fix this?

Thanks for your help!

P.S i've updated my PR with this change.

oliviabrown9 commented:
Haven't had too much time to dig into this one yet (will look more into it later this week), but the thing I find most interesting is that the current state of this PR doesn't show the animation the first time you click to edit. However, if you cancel the editing and then edit again, the animation is there again.

oliviabrown9 commented:
CCing @sblatz to take over reviewing.

sblatz commented:
@marchiore Hey! Thanks for your work on this. Would you be able to pull from 7.0 dev so I can see exactly what code has changed? I think some of the diffs are messed up right now.

marchiore commented:
@sblatz Done!

I accidentally closes this PR, so I create another one (#1359) with the changes I made.
